src/Microsoft.OpenApi/Reader/OpenApiJsonReader.cs

```diff
@@ -162,8 +157,8 @@ public T ReadFragment<T>(JsonNode input,
             IOpenApiElement element = null;
             try
             {
-                // Parse the OpenAPI element
-                element = context.ParseFragment<T>(input, version);
+                // Parse the OpenAPI element asynchronously
+                element = await Task.Run(() => context.ParseFragment<T>(input, version));
```

Comment:
why the extra Task.Run here?

Reply:
The `context.ParseFragment` method is synchronous hence the Task.Run to run it async and avoid blocking the calling thread

Reply:
right, but all of the work this method is doing is CPU bound? (no IOs)

Reply:
Yes you're right. Let me refactor that to remove the async operation.